python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -425,75 +424,45 @@ def _convert_to_sampled_subgraph(
             # The sampled graph is already a homogeneous graph.
             sampled_csc = CSCFormatBase(indptr=indptr, indices=indices)
         else:
```

Comment:
Right at the beginning of the else block, can we add this line:
`node_type_offset = self.node_type_offset.to(column.device)`

For GPU sampling, when we pin the graph, the node_type_offset is on the CPU. We need to temporarily move it to the GPU and use the moved version in the code below.

Reply:
> Right at the beginning of the else block (inside else), can we add this line: `self.node_type_offset = self.node_type_offset.to(column.device)`
> 
> For GPU sampling, when we pin the graph, the `self.node_type_offset` is on the CPU. We need to move it to the GPU before the following code starts to run. After the initial movement, the later calls will not do anything as it will already be on the device.

Make sense. The line has been added.